### PR TITLE
feat: add brew install for lula

### LINKED ISSUE #702: Brew Install Lula
### Is your feature request related to a problem? Please describe.

Add the setup to implement Brew installations of the Lula Binary in Lula's release process.

### Describe the solution you'd like

When I run brew install defenseunicorns/lula I would like to have the Lula binary installed.

### Additional context

Lula release process already supports source installs and several other Product tools support Brew installations.

### Expected Deliverable

- [ ] Brew install

### PR DESCRIPTION
**Pull request by CloudBeard**
## Description

Added Brew release configuration. This should create the release inside of the Defense Unicorns Brew repo. Permissions were previously setup. 

A separate issues exists for updating the documentation after a release has been cut and testing has bee conducted. https://github.com/defenseunicorns/lula/issues/321

This issue would be a follow on after a release has been made.

## Related Issue

Fixes #
#702 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Schema Updates](https://github.com/defenseunicorns/lula/blob/main/docs/community-and-contribution/schema-updates.md) applied
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/lula/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

File: .goreleaser.yaml [hunk 1]
```diff
@@ -42,6 +42,41 @@ snapshot:
...
+      token: "{{ .Env.HOMEBREW_TAP_GITHUB_TOKEN }}"
+      branch: "{{ .ProjectName }}-{{ .Tag }}"
+      pull_request:
+        enabled: true
+        base:
+          branch: main
+          owner: defenseunicorns
+          name: homebrew-tap
+    commit_msg_template: "Brew formula update for {{ .ProjectName }} version {{ .Tag }}"
+    homepage: "https://github.com/defenseunicorns/lula"
```
Comment (brandtkeller) [inline .goreleaser.yaml]: Future Proof a little
```suggestion
    homepage: "https://lula.dev"
```

File: .goreleaser.yaml [hunk 2]
```diff
@@ -42,6 +42,41 @@ snapshot:
...
+      token: "{{ .Env.HOMEBREW_TAP_GITHUB_TOKEN }}"
+      branch: "{{ .ProjectName }}-{{ .Tag }}"
+      pull_request:
+        enabled: true
+        base:
+          branch: main
+          owner: defenseunicorns
+          name: homebrew-tap
+    commit_msg_template: "Brew formula update for {{ .ProjectName }} versioned release {{ .Tag }}"
+    homepage: "https://github.com/defenseunicorns/lula"
```
Comment (brandtkeller) [inline .goreleaser.yaml]:
```suggestion
    homepage: "https://lula.dev"
```

**Review by brandtkeller — CHANGES_REQUESTED** [review]
Minor future proofing commands but i'd also like to make a request:

`goreleaser check` is returning errors and deprecations. While we're here I'd like to expand scope a small amount such that we have a repeatable process for these kinds of updates. 

- Ensure the configuration is valid per `goreleaser check`
- Document these checks in our release process (if applicable)

Besides that - as a practice I'd like to start capturing "how does a reviewer perform review of these changes" in our PR descriptions -> I'll work this into a PR template.

File: .goreleaser.yaml (hunk 1 above)
Comment (CloudBeard) [reply .goreleaser.yaml]: Updated

File: .goreleaser.yaml (hunk 2 above)
Comment (CloudBeard) [reply .goreleaser.yaml]: Updated

**Comment by CloudBeard** [comment]
Added a GitHub Action that will run GoReleaser check on any pushes or PRs that contain a change to the .goreleaser.yaml file.

Also tweaked the wording a bit in our release documentation to call out the brew info and the action. We don't go too in-depth into distros.

File: docs/community-and-contribution/release-process.md [hunk 3]
```diff
@@ -20,7 +20,8 @@ The most important prefixes you should have in mind are:
 
 ### How can I influence the version number for a release?
 
-PR titles should also follow this pattern and are linted using [commitlint](https://commitlint.js.org/). The PR title will determine the version bump. When a PR is merged (squashed) release-please will kick off a release PR. When that release PR is approved and merged, release-please will create a draft release. Once that draft release is published go-releaser with build and publish the assets.  
+PR titles should also follow this pattern and are linted using [commitlint](https://commitlint.js.org/). The PR title will determine the version bump. When a PR is merged (squashed) release-please will kick off a release PR. When that release PR is approved and merged, release-please will create a draft release. Once that draft release is published go-releaser with build and publish the assets,including creating a release in our Homebrew tap repository: [https://github.com/defenseunicorns/homebrew-tap](https://github.com/defenseunicorns/homebrew-tap)
```
Comment (brandtkeller) [inline docs/community-and-contribution/release-process.md]:
```suggestion
PR titles should also follow this pattern and are linted using [commitlint](https://commitlint.js.org/). The PR title will determine the version bump. When a PR is merged (squashed) release-please will kick off a release PR. When that release PR is approved and merged, release-please will create a draft release. Once that draft release is published go-releaser with build and publish the assets, including creating a release in our Homebrew tap repository: [https://github.com/defenseunicorns/homebrew-tap](https://github.com/defenseunicorns/homebrew-tap)
```

**Review by brandtkeller — APPROVED** [review]

**Pull request merged**